### ISSUE TITLE
 Notepad++ hangs and then unable to click on notepad++ window anywhere. #4295

### ISSUE THREAD

**Issue by pinodev** [wakatime/notepadpp-wakatime#34]
Hi there,

Please see ticket https://github.com/notepad-plus-plus/notepad-plus-plus/issues/4295, originally logged for Notepad++, but it turns out it is the time keeping plugins that mess with the application.

Is there a way to debug a plugin ?
(I mean, if I compile the plugin myself and I run Notepad++, and later see it freeze, would I be able to break in and debug to see where it's stuck ?)

Kind regards,

P.

**Comment by p0358** [comment]
According to recently closed #27 #26, they claimed to had just fixed this issue. I will need to review the changes they made there and incorporate them into my fork. It's probably NPP plugin platform for C# update that fixed the issues, but it's just my guess for now. @pinodev can you check if this issue persists for you with latest version of WakaTime alone? That way we could see if they actually fixed it.

**Comment by pinodev** [comment]
I have installed (the very latest version of) WakaTime, will try it alone and will let you know if I'm still experiencing the issue. Speak later. P.

**Comment by gandarez** [comment]
@pinodev news?

**Comment by p0358** [comment]
Seems that updating the plugin platform to v0.94.00 alone didn't fix the freezing in my fork, I just encountered it again now
EDIT: But I think I may have a track on why my one is freezing, I need to test a bit further. But it would be valid only if current WakaTime doesn't freeze anymore as well
@pinodev btw, it can be more-less debugged according to instructions here: https://github.com/kbilsted/NotepadPlusPlusPluginPack.Net#how-to-debug-plugins

**Comment by pinodev** [comment]
> 
> 
> @pinodev news?

@gandarez 

I experienced with WakaTime _on its own_ for 3 good days, and I didn't get the freeze. So I reinstalled CodeStats (alongside WakaTime, which I kept). And everything worked fine for a few more days. (Which is already great news because when I originally logged this ticket two weeks ago, I was getting the freeze at least once per hour.). Oh, and btw, I also moved from using the conventional Notepad++ Setup (in Program Files), to Portable mode.

And just as I though the issue was somehow totally fixed, I encountered it again, but only once, two days ago. But I can't tell which plugin caused it. Also, I can't tell if running the binaries from a portable location instead of Program Files helped in getting less of these problems (I would have to move back to using the conventional Setup and see if I'm getting the freeze on a regular basis again).

Anyway, there might be something in the environment triggering the error: either local file permissions, or local paths (eg. running from "Program Files" vs. running from a portable location), or maybe something to do with the network (such as being connected to a VPN), unless it has to do with the Notepad++ session itself (eg. the files currently being edited)... I didn't see this happening enough times and I didn't try enough configurations to be able to find a pattern.

Also I'm still hoping that when it happens, I would be able to attach the process and see where it's getting stuck (I compiled both plugins for that purpose, so that I could potentially step through the source).

@p0358 - thanks for the link, I'll make sure that I can debug the plugin in normal conditions, so that when the freeze occurs, I could potentially break execution and see what does the callstack look like.
